warehouse/postgres/postgres.go

```diff
@@ -0,0 +1,520 @@
+package postgres
+
+import (
+	"compress/gzip"
+	"database/sql"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"github.com/lib/pq"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/rruntime"
+	"github.com/rudderlabs/rudder-server/services/filemanager"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+	uuid "github.com/satori/go.uuid"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+var (
+	warehouseUploadsTable string
+	stagingTablePrefix    string
+	maxParallelLoads      int
+)
+
+const (
+	host     = "host"
+	dbName   = "database"
+	user     = "user"
+	password = "password"
+	port     = "port"
+	sslMode  = "sslMode"
+)
+
+var dataTypesMap = map[string]string{
+	"int":      "bigint",
+	"float":    "numeric",
+	"string":   "text",
+	"datetime": "timestamp with time zone",
```

Comment:
should this be timestamp with time zone? I think we are setting it as timestamp in other warehouses and it is equivalent to timestamp without time zone in them. Do verify in others also if we are setting equivalent to timestamp without time zone